### PR TITLE
luminous: rgw: lifecycle feature

### PR DESCRIPTION
**Pull request by mattbenjamin**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by smithfarm** [comment]
Is all of this stuff already in mimic?

**Comment by theanalyst** [comment]
Note https://tracker.ceph.com/issues/39654

**Comment by yuriw** [comment]
[wip-yuri5-testing-2019-06-11-0036-luminous](https://trello.com/c/dOCwVIFT/636-wip-yuri5-testing-2019-06-11-0036-luminous)

**Comment by theanalyst** [comment]
I guess we need to pull in https://github.com/ceph/ceph/pull/28049 in order to not introduce a regression in RGW Put LC

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ std::string rgwlc_s3_expiration_header(
     char exp_buf[100];
     time_t exp = ceph::real_clock::to_time_t(*expiration_date);
     if (std::strftime(exp_buf, sizeof(exp_buf),
-		      "%c", std::gmtime(&exp))) {
+		      "%a, %d %b %y %T %z", std::gmtime(&exp))) {
```
Comment (beomseok-park) [inline src/rgw/rgw_lc.cc]: see https://github.com/ceph/ceph/pull/26160#discussion_r293692570

**Comment by smithfarm** [comment]
luminous EOL

**Pull request closed without merging**